fv3core/stencils/a2b_ord4.py

```diff
@@ -22,485 +36,435 @@
+
+        # compute_qyy
+        qyy = lagrange_x(qy)
+        with horizontal(region[i_start + 1, :]):
+            qyy = cubic_interpolation_west(qy, qout, qyy)
+        with horizontal(region[i_end, :]):
+            qyy = cubic_interpolation_east(qy, qout, qyy)
+
+        with horizontal(region[i_start + 1 : i_end + 1, j_start + 1 : j_end + 1]):
+            qout = 0.5 * (qxx + qyy)
+        # }
+
+        if __INLINED(REPLACE):
+            qin = qout
+
+
+def _make_grid_storage_2d(grid_array: gt4py.storage.Storage, index: int = 0):
+    grid = spec.grid
+    return gt4py.storage.from_array(
+        grid_array[:, :, index],
+        backend=global_config.get_backend(),
+        default_origin=grid.compute_origin()[:-1],
+        shape=grid_array[:, :, index].shape,
+        dtype=grid_array.dtype,
+        mask=(True, True, False),
     )
 
 
-def compute(qin, qout, kstart=0, nk=None, replace=False):
+def compute(
+    qin: FloatField,
+    qout: FloatField,
+    kstart: int = 0,
+    nk: Optional[int] = None,
+    replace: bool = False,
+):
+    grid = spec.grid
     if nk is None:
-        nk = grid().npz - kstart
-    extrapolate_corners(qin, qout, kstart, nk)
-    if spec.namelist.grid_type < 3:
-        compute_qout_edges(qin, qout, kstart, nk)
-        qx = compute_qx(qin, qout, kstart, nk)
-        qy = compute_qy(qin, qout, kstart, nk)
-        qxx = compute_qxx(qx, qout, kstart, nk)
-        qyy = compute_qyy(qy, qout, kstart, nk)
-        compute_qout(qxx, qyy, qout, kstart, nk)
-        if replace:
-            copy_stencil(
-                qout,
-                qin,
-                origin=(grid().is_, grid().js, kstart),
-                domain=(grid().ie - grid().is_ + 2, grid().je - grid().js + 2, nk),
-            )
-    else:
-        raise Exception("grid_type >= 3 is not implemented")
+        nk = grid.npz - kstart
+    agrid1 = _make_grid_storage_2d(grid.agrid1)
+    agrid2 = _make_grid_storage_2d(grid.agrid2)
+    bgrid1 = _make_grid_storage_2d(grid.bgrid1)
+    bgrid2 = _make_grid_storage_2d(grid.bgrid2)
+    dxa = _make_grid_storage_2d(grid.dxa)
+    dya = _make_grid_storage_2d(grid.dya)
+    edge_n = _make_grid_storage_2d(grid.edge_n)
+    edge_s = _make_grid_storage_2d(grid.edge_s)
+    edge_e = _make_grid_storage_2d(grid.edge_e)
+    edge_w = _make_grid_storage_2d(grid.edge_w)
+
+    stencil = gtstencil(definition=_a2b_ord4_stencil, externals={"REPLACE": replace})
+
+    stencil(
+        qin,
+        qout,
+        agrid1,
+        agrid2,
+        bgrid1,
+        bgrid2,
+        dxa,
+        dya,
+        edge_n,
+        edge_s,
+        edge_e,
+        edge_w,
+        origin=(grid.is_, grid.js, kstart),
+        domain=(grid.nic + 1, grid.njc + 1, nk),
+    )
+
+    # if replace:
+    #     copy_stencil(
+    #         qout,
+    #         qin,
+    #         origin=(grid.is_, grid.js, kstart),
+    #         domain=(grid.nic + 1, grid.njc + 1, nk),
+    #     )
```

Comment:
Is this intended to be commented out or was that for debugging?

Reply:
Oh I see now, can be deleted, awesome

Reply:
Ah, I forgot to delete it after adding back the external. Good catch!